src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +142,207 @@ async function uploadWithProgress({
     )
 }
 
-function statFile(file: vscode.Uri) {
-    return statSync(file.fsPath).size
+interface BucketQuickPickItem extends vscode.QuickPickItem {
+    bucket: S3.Bucket | undefined
 }
 
-async function refreshNode(node: S3BucketNode | S3FolderNode, commands: Commands): Promise<void> {
-    node.clearChildren()
-    return commands.execute('aws.refreshAwsExplorerNode', node)
+/**
+ * Will display a quick pick with the list of all buckets owned by the user.
+ * 
+ * @returns Bucket selected by the user, undefined if no bucket was selected or the quick pick was cancelled.
+ * 
+ * @throws Error if there is an error calling s3
+ */
+export async function promptUserForBucket(
```

Comment:
`src/lambda/wizards/samDeployWizard.ts:S3_BUCKET` has a bucket prompt that allows (1) creating a new bucket, (2) entering bucket by name (no list), (3) selecting from list (if user has list permissions), (4) select recently-used buckets.

Related PRs:
- https://github.com/aws/aws-toolkit-vscode/pull/1483
- https://github.com/aws/aws-toolkit-vscode/pull/1527

We should lift that logic out of src/lambda/wizards/samDeployWizard.ts so that we can use it whenever a bucket name is needed. Maybe in a separate PR.

Reply:
That can come in very useful